src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_TodoComments.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.TodoComments;
+using Roslyn.Utilities;
+using RoslynLogger = Microsoft.CodeAnalysis.Internal.Log.Logger;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    // root level service for all Roslyn services
+    internal partial class CodeAnalysisService : IRemoteTodoCommentService
+    {
+        /// <summary>
+        /// This is top level entry point for TodoComments service from client (VS).
+        /// 
+        /// This will be called by ServiceHub/JsonRpc framework
+        /// </summary>
+        public async Task<IList<TodoComment>> GetTodoCommentsAsync(DocumentId documentId, ImmutableArray<TodoCommentDescriptor> tokens)
+        {
+            using (RoslynLogger.LogBlock(FunctionId.CodeAnalysisService_GetTodoCommentsAsync, documentId.ProjectId.DebugName, CancellationToken))
+            {
+                try
+                {
+                    var solution = await GetSolutionAsync().ConfigureAwait(false);
+                    var document = solution.GetDocument(documentId);
+
+                    var service = document.GetLanguageService<ITodoCommentService>();
```

Comment:
I already have method that return whether language is supported in OOP or not. since OOP can only support language that has ILanguageService MEF exported in OOP. right now, it only compose C# and VB to MEF container so other lanugages are not supported until explicitly added to OOP MEF composition.